include/xcshell/redirectelement.h

```diff
@@ -0,0 +1,16 @@
+#ifndef XCSHELL_REDIRECTELEMENT_H
+#define XCSHELL_REDIRECTELEMENT_H
+
+#include <string>
+#include <utility>
+#include <vector>
+
+struct RedirectElement {
```

Comment:
Maybe `CommandParseResult` or some other name more make sense? Because this struct is not only for redirect, it's also contains command and args user want to execute.